main.go

```diff
@@ -281,17 +494,28 @@ func (c *controller) saveHashring(hashring []receive.HashringConfig) error {
 		BinaryData: nil,
 	}
 
+	c.configmapChangeAttempts.Inc()
```

Comment:
I have decided to keep attempts and errors (with types). I don't have enough domain knowledge to anticipate if this has any downsides. I could also track API reach out with type. What do you think?